packages/editor/src/components/post-author/index.js

```diff
@@ -5,45 +5,159 @@ import { __ } from '@wordpress/i18n';
+			return;
+		}
+
+		this.requestResults( query, populateResults );
+	}
+
+	resolveResults( results ) {
+		this.authors = results;
+		return results.map( ( author ) => ( author.name ) );
+	}
+
+	getCurrentAuthor( authorId ) {
+		if ( ! authorId ) {
+			return;
+		}
+		apiFetch( { path: '/wp/v2/users/' + encodeURIComponent( authorId ) } ).then( ( results ) => {
+			this.setState( { postAuthor: results } );
+		} );
 	}
 
-	setAuthorId( event ) {
+	setAuthorId( selection ) {
+		if ( ! selection ) {
+			return;
+		}
 		const { onUpdateAuthor } = this.props;
-		const { value } = event.target;
-		onUpdateAuthor( Number( value ) );
+		if ( typeof selection === 'string' ) {
+			// Author name from the autocompleter.
+			const author = this.authors.find( ( singleAuthor ) => {
+				return singleAuthor.name === selection;
+			} );
+			onUpdateAuthor( Number( author.id ) );
+		} else {
+			// Author ID from the select.
+			onUpdateAuthor( Number( selection.target.value ) );
+		}
 	}
 
 	render() {
-		const { postAuthor, instanceId, authors } = this.props;
-		const selectId = 'post-author-selector-' + instanceId;
-
-		// Disable reason: A select with an onchange throws a warning
+		const { postAuthor } = this.state;
+		const { postAuthorId, instanceId, authors } = this.props;
+		const selectId = `post-author-selector-${ instanceId }`;
+		let selector;
+		if ( ! postAuthor ) {
+			return null;
+		}
 
-		/* eslint-disable jsx-a11y/no-onchange */
-		return (
-			<PostAuthorCheck>
-				<label htmlFor={ selectId }>{ __( 'Author' ) }</label>
+		if ( authors.length < 50 ) {
+			/* eslint-disable jsx-a11y/no-onchange */
+			selector =
 				<select
 					id={ selectId }
-					value={ postAuthor }
-					onChange={ this.setAuthorId }
+					value={ postAuthorId }
 					className="editor-post-author__select"
+					onChange={ this.setAuthorId }
 				>
 					{ authors.map( ( author ) => (
 						<option key={ author.id } value={ author.id }>{ author.name }</option>
 					) ) }
-				</select>
+				</select>;
+		} else {
+			selector =
+				<Autocomplete
+					id={ selectId }
+					minLength={ 2 }
+					showAllValues={ true }
+					autoselect={ true }
+					defaultValue={ postAuthor ? postAuthor.name : '' }
+					displayMenu="overlay"
+					onConfirm={ this.setAuthorId }
+					source={ this.suggestAuthor }
+					showNoResultsFound={ false }
+					tStatusQueryTooShort={ ( minQueryLength ) =>
+						__( `Type in ${ minQueryLength } or more characters for results` ) }
+					tStatusNoResults={ () => __( 'No search results' ) }
+					tStatusSelectedOption={ ( selectedOption, length ) => __( `${ selectedOption } (1 of ${ length }) is selected` ) }
+					tStatusResults={ ( length, contentSelectedOption ) => {
+						const words = {
+							result: ( length === 1 ) ? __( 'result' ) : __( 'results' ),
+							is: ( length === 1 ) ? __( 'is' ) : __( 'are' ),
+						};
+						return <span>{ length } { words.result } { words.is } available. { contentSelectedOption }</span>;
```

Comment:
And this using `_n`

Reply:
ah, ok! thanks for the feedback @TimothyBJacobs!

Reply:
Thanks for this suggestion, changed in https://github.com/WordPress/gutenberg/pull/7385/commits/1d13ad8a9ffc2aae38d43d17464cf66b65fe8459